app/pb_kits/playbook/pb_gauge/_gauge.jsx

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+import React, { useEffect, useRef } from 'react'
+import { pbChart } from '../'
+import { spacing } from '../utilities/spacing.js'
+import classnames from 'classnames'
+import Highcharts from 'highcharts'
+
+import {
+  buildAriaProps,
+  buildCss,
+  buildDataProps,
+} from '../utilities/props'
+
+type GaugeProps = {
+  aria: Object,
+  className?: String,
+  chartData?: Array,
+  data?: Object,
+  fullCircle: Boolean,
+  height: String,
+  id?: String,
+  max: Number,
+  min: Number,
+  prefix: String,
+  showLabels: Boolean,
+  style: String,
+  suffix: String,
+  title: String,
+  tooltipHtml: String,
+}
+
+const Gauge = (props: GaugeProps) => {
+  const {
+    aria = {},
+    className,
+    chartData = [{ name: 'Name', value: 0 }],
+    data = {},
+    fullCircle = false,
+    height = null,
+    id,
+    max = 100,
+    min = 0,
+    prefix = '',
+    showLabels = false,
+    style = 'solidgauge',
+    suffix = '',
+    title = '',
+    tooltipHtml = '<span style="font-weight: bold; color:{point.color};">●</span>{point.name}: ' + '<b>{point.y}</b>',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
+  const dataProps = buildDataProps(data)
+
+  const css = buildCss({
+    'pb_gauge_kit': true,
+  })
+  // Runs first time component Renders
+  useEffect(() => {
+    chartData.forEach((obj) => {
+      obj.y = obj.value
+      delete obj.value
+    })
+
+    new pbChart('.selector', {
```

Comment:
Is this an external dom manipulation?

Reply:
External to React, yes.  It instantiates a pbChart obj defined in [this file here](https://github.com/powerhome/playbook/blob/9c853a9c306b453deab4d0c1c4943d5c8999d4fe/app/pb_kits/playbook/plugins/pb_chart.js).  It relies on the [Highcharts library](https://www.highcharts.com/demo/gauge-speedometer) which in turn manipulates the dom.  Tried to use the Highcharts api to directly affect/update the chart and simply pass React state to it (to avoid unexpected Highcharts behavior when React re-renders).

Reply:
Have you considered https://github.com/highcharts/highcharts-react ? It's not a good idea to mess with dom manipulation as it's react job to rewrite that whenever it feels it should.

Reply:
Had taken a look at highcharts-react.  It was a breeze to setup in codesandbox.  Ultimately we want to move all our react Highcharts kits (Bar Graph, Circle Chart, Line Graph, Circle Chart) over to use highcharts-react but that effort was outside the scope of this project.  

Agreed though, I ran into many of the issues why it's not a good idea to mess with dom manipulation outside of React's concern as I was building the kit.  Long term we will likely be migrating React kits over to react-highcharts.

Reply:
If that's outside `React`'s context, where will that `selector` class be defined exactly? I guess I'm trying to understand how [`this.element`](https://github.com/powerhome/playbook/blob/9c853a9c306b453deab4d0c1c4943d5c8999d4fe/app/pb_kits/playbook/plugins/pb_chart.js#L30) fits into the implementation.

Reply:
Much of pbChart was inherited code, so I'm not entirely what the purpose of the `'.selector'` class is.  Looks like the instance variable `element` is never even used?

Reply:
that's what I was trying to pinpoint :) not sure either.

Reply:
@jasperfurniss do you know what the `'.selector'` argument in pbChart's constructor does?  It's in there for all highcharts instances in Playbook.  I copy and pasted it when setting up the Gauge kit and never got around to investigating.  Maybe it was originally created to assign a selector to an outer div?  Looks like it was never implemented.

Reply:
@bh247484 Just now seeing this in my notifications. 
I don't believe there is a purpose for the `'.selector'` that I'm aware of. I considered removing it myself. We would need to test, and do some digging to see what it actually does or was intended to do.